### PR TITLE
feat: queries to add and remove single or multiple trees

### PR DESCRIPTION
**Pull request by AlexWang05**
### Description
Query support for invoking existing Supabase Database Functions for the following actions:
- Add a single tree given a species, using a random uuid and default values for the rest of the properties.
- Add multiple trees given a species and quantity.
- Remove a tree given a uuid.
- Remove multiple trees given a list of uuids.

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Examine `supabase/queries.ts` and use of `client.ts` for Supabase queries. Check compliance with Linear sprint specs.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

## Testing
All queries and Supabase Functions are tested manually, through invoking buttons written to trigger queries in `App.tsx`. Adding single and multiple trees, and removing single and multiple trees (UUID JSON) worked as intended,and is reflected in the database.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
N/A

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
N/A

CC: @christophertorres1

### CODE REVIEW

File: src/supabase/queries.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+export async function addTree(species: string) {
+  const { error } = await supabase.rpc('add_tree', { species });
+
+  if (error) {
+    throw new Error(`Error adding tree: ${error.message}`);
+  }
+}
+
+// Function to add multiple trees
+export async function addMultipleTrees(
```
Comment (adityapawar1) [inline src/supabase/queries.ts:13]: When I run `addMultipleTrees([{ species: 'Big Tree', quantity: 4 }])`(lmk if i'm calling the function wrong) i'm seeing `Error adding multiple trees: Could not find the function public.add_multiple_trees(trees) in the schema cache`. I think it could be bc the db function takes in species and quantity separately and this function only passes in a `trees` argument? I'm not too familiar with db functions so idk

**Review by adityapawar1 — CHANGES_REQUESTED** [review]
I think there is one small bug with addMultiple trees (see the other review comment), but other than that it looks good!

File: src/supabase/queries.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  trees: { species: string; quantity: number }[],
+) {
+  const { error } = await supabase.rpc('add_multiple_trees', {
```
Comment (christophertorres1) [inline src/supabase/queries.ts:16]: You can pass in the two arguments separately instead of using JSON.stringify here.

File: src/supabase/queries.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  if (error) {
+    throw new Error(`Error removing tree: ${error.message}`);
+  }
+}
+
+// Function to remove multiple trees by a list of UUIDs
+export async function removeMultipleTrees(treeIds: string[]) {
+  const { error } = await supabase.rpc('remove_multiple_trees', {
+    p_tree_ids: treeIds,
```
Comment (christophertorres1) [inline src/supabase/queries.ts:39]: Consider renaming to 'tree_uuids' to keep things consistent with removeTree.

**Review by christophertorres1 — CHANGES_REQUESTED** [review]
Minor changes but wonderful job overall!

File: src/supabase/queries.ts (hunk 1 above)
Comment (AlexWang05) [reply src/supabase/queries.ts]: resolved via refactoring
